agent/engine/docker_task_engine.go

```diff
@@ -1133,15 +1135,18 @@ func (engine *DockerTaskEngine) createContainer(task *apitask.Task, container *a
 	}
 
 	if execcmd.IsExecEnabledContainer(container) {
-		tId, err := task.GetID()
-		if err != nil {
-			herr := &apierrors.HostConfigError{Msg: err.Error()}
-			return dockerapi.DockerContainerMetadata{Error: apierrors.NamedError(herr)}
+		tID, err := task.GetID()
+		if err == nil {
+			err = engine.execCmdMgr.InitializeContainer(tID, container, hostConfig)
 		}
-		err = engine.execCmdMgr.InitializeContainer(tId, container, hostConfig)
 		if err != nil {
-			herr := &apierrors.HostConfigError{Msg: err.Error()}
-			return dockerapi.DockerContainerMetadata{Error: apierrors.NamedError(herr)}
+			// TODO: Send a managed agent state change event for this
+			seelog.Warnf("Exec Agent initialization: %v . Continuing to start container without enabling exec feature.", err)
+			container.UpdateManagedAgentByName(execcmd.ExecuteCommandAgentName, apicontainer.ManagedAgentState{
```

Comment:
I think it'd be better to update the managed agent inside of `execCmdMgr.InitializeContainer`. That way all the initialization logic remains in one place.

Since InitializeContainer has many error paths, I think we would benefit from golang's named return parameters, so that we invoke `UpdateManagedAgentByName` only once.

e.g.

```golang
func (m *manager) InitializeContainer(taskId string, container *apicontainer.Container, hostConfig *dockercontainer.HostConfig) (rErr error) {
	defer func() {
                // If this function returns error, update managed agent accordingly 
		if rErr != nil {
			container.UpdateManagedAgentByName(execcmd.ExecuteCommandAgentName, apicontainer.ManagedAgentState{
				InitFailed: true,
				Status:     apicontainerstatus.ManagedAgentStopped,
				Reason:     err.Error(),
			})
		}
	}()
	if tID == "" {
		return errors.New("Task id cannot be blank")
	}
	.
	.
	.
}
```

Reply:
Thanks for the idea. I changed it.